autosklearn/data/validation.py

```diff
@@ -84,9 +84,11 @@ def validate_features(
         """
 
         # Pre-process dataframe to make them numerical
-        if hasattr(X, "iloc"):
+        # Also, encode numpy categorical objects
+        if (hasattr(X, "iloc") or (hasattr(X, "dtype") and X.dtype.type is np.str_)) \
```

Comment:
Could you please let me know when the 2nd part of the or statement is triggered?

Reply:
Just seeing that this is for numpy 'categorical'. I think we should simply not support non-number types when the user passes in numpy arrays.

Reply:
Understood. I will add a explanatory message with this request, that we only support numpy numerical types